src/QMCWaveFunctions/Fermion/MultiSlaterDetTableMethod.h

```diff
@@ -186,6 +187,14 @@ class MultiSlaterDetTableMethod : public WaveFunctionComponent, public Optimizab
 
   void evaluateDerivativesWF(ParticleSet& P, const opt_variables_type& optvars, Vector<ValueType>& dlogpsi) override;
 
+  /** Compute ratios of the individual Slater determinants and the total MSD value.  
+   *  These are obtained via derivatives of the log of the wavefunction component with respect to the optimizable parameters.
+   *  Unlike evaluateDerivativesWF, knowledge of mappings from component parameters to the full set of wavefunction parameters is not required.
+   */
+  void detRatios(Vector<ValueType>& ratios);
```

Comment:
This function doesn't conform naming rules that functions start with a verb in lower case. I would recommend `calcIndividualDetRatios`.
Although calculating individual determinant ratios and calculating parameter derivatives share very similar internal implementation in the case of CI not necessarily the case with CSFs. They are conceptually different from APIs perspective. The difference is larger than just a mapping for example the det 0 is skipped when calculating derivatives. So I would recommend removing
```
 These are obtained via derivatives of the log of the wavefunction component with respect to the optimizable parameters.
   *  Unlike evaluateDerivativesWF, knowledge of mappings from component parameters to the full set of wavefunction parameters is not required.`
```
in the documentation. Please also document the argument `ratios`.

Reply:
Also please test this function at
```
https://github.com/QMCPACK/qmcpack/blob/2764f973212f911490ff488a44ed011b22678154/src/QMCWaveFunctions/tests/test_multi_slater_determinant.cpp#L112
```
although you'll need to findMSD and cast the type to MSD first.
Please test the first element and a non-first element.